language_service/src/hover.rs

```diff
@@ -204,26 +300,68 @@ impl Visitor<'_> for HoverVisitor<'_> {
     }
 }
 
-fn markdown_with_doc(doc: &str, namespace: Option<Rc<str>>, code: impl Display) -> String {
-    let parsed_doc = parse_doc(doc);
+fn protocol_span(span: qsc::Span) -> protocol::Span {
```

Comment:
This doesn't matter in practice today since we only deal with single files, but... technically, `span.lo` and `span.hi` are offsets into the `SourceMap`, whereas `protocol::Span` expects an offset into the current source. Once we move to the "multiple file world" we're going to notice the offsets start being off. To get ahead of that, I'd suggest you take the `SourceMap` here and use `find_by_offset` to get the correct source offset and subtract it. (definition.rs does this correctly today, as an example).

Reply:
Oh that is a good call out. That would have been tricky to debug down the line. I've updated the code to use source offsets. Thanks!